pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -1229,6 +1229,26 @@ self: super: {
   # Tests require newer package version.
   aeson_1_5_6_0 = dontCheck (doJailbreak super.aeson_1_5_6_0);
 
+  aeson_2_2_0_0 = super.aeson_2_2_0_0.override {
+    th-abstraction = self.th-abstraction_0_5_0_0;
+  };
+
+  generic-deriving = super.generic-deriving.override {
+    th-abstraction = self.th-abstraction_0_5_0_0;
+  };
+
+  th-lift = super.th-lift.override {
+    th-abstraction = self.th-abstraction_0_5_0_0;
+  };
+
+  th-expand-syns = super.th-expand-syns.override {
+    th-abstraction = self.th-abstraction_0_5_0_0;
+  };
+
+  bifunctors = super.bifunctors.override {
+    th-abstraction = self.th-abstraction_0_5_0_0;
+  };
```

Comment:
Since things like `generic-deriving`, `th-lift`, etc already compile using the current version of `th-abstraction`, we shouldn't override the `th-abstraction` version in all of these packages (since that will make them incompatible with any packages _not_ overriding the `th-abstraction` version).

If I understand correctly, your goal with this is to get `spago` compiling correctly, right?

In that case, it would probably be better to throw `th-abstraction` into the `.overrideScope` for `spago` in `pkgs/development/haskell-modules/configuration-nix.nix`.  That way, these changes would only affect spago (and not the rest of the package set).

Also, can you make sure your commit message and PR title matches the format specified in https://github.com/NixOS/nixpkgs/blob/9108a20782535741433c304f6a4376cb8b364b89/CONTRIBUTING.md#commit-conventions?

Reply:
Thanks, yes, I am attempting to fix Spago: https://github.com/NixOS/nixpkgs/issues/253198

However, I am also interested in specifically fixing the `aeson_2_2_0_0` package, regardless of Spago.

Reply:
> However, I am also interested in specifically fixing the aeson_2_2_0_0 package, regardless of Spago.

In that case, maybe using an `.overrideScope` on `aeson_2_2_0_0` for `th-abstraction_0_5_0_0` would be the simplest solution.